python/ray/util/scheduling_strategies.py

```diff
@@ -0,0 +1,114 @@
+from typing import Union, Optional
+from ray.util.annotations import PublicAPI
+from ray.util.placement_group import PlacementGroup
+
+
+@PublicAPI(stability="beta")
+class SchedulingStrategy(object):
+    """Base class for all scheduling strategies.
+    """
+
+    def to_dict(self) -> dict:
+        """Convert this scheduling strategy into a dict for purposes of json
+        serialization.
+
+        Used when passing a scheduling strategy as an option to a
+        Ray client remote function.
+        See set_task_options in util/client/common.py.
+
+        Return:
+            Json-serializable dictionary representing the scheduling strategy.
+        """
+        raise NotImplementedError("to_dict() has to be implemented")
+
+    @staticmethod
+    def from_dict(ss_dict: dict) -> "SchedulingStrategy":
```

Comment:
Is there a reason to prefer JSON instead of just pickling this class for Ray client? It seems to add quite a bit of cruft.